src/HandHistory.js

```diff
@@ -0,0 +1,148 @@
+class HandHistory {
+
+    constructor(player, cards) {
+        this.actions = [];
+        this.hero = player;
+        this.heroCards = cards;
+    }
+
+    // After you build one then for fuck's sake don't modify it. I'm trusting you.
```

Comment:
If you wanted to enforce that & conveniently have React re-render the component when the HandHistory changed, you could do the fluent interface idea & have it return a new HandHistory instead of simply modifying the existing one. (React uses object identity to decide whether to re-render.)